sdk/core/core-lro/src/poller.ts

```diff
@@ -325,11 +324,11 @@ export abstract class Poller<TState extends PollOperationState<TResult>, TResult
           // we are not expecting a result anyway. To assert that we might not
           // have a result eventually after finishing polling, we cast the result
           // to TResult.
-          this.resolve(state.result as TResult);
+          this.resolve(this.operation.state.result as TResult);
```

Comment:
Why is this a bug? let's sync up

Reply:
Oh got it! make sense. Thank you!

Reply:
We synced up offline. grabbing a handle on `this.operation.state` can cause stale data to be returned if the update function reassigns state